crates/utils/re_arrow_util/src/arrow_util.rs

```diff
@@ -38,6 +32,13 @@ pub fn into_arrow_ref(array: impl Array + 'static) -> ArrayRef {
     std::sync::Arc::new(array)
 }
 
+/// Returns an iterator with the lengths of the offsets
+pub fn offsets_lengths(
+    offsets: &arrow::buffer::OffsetBuffer<i32>,
+) -> impl Iterator<Item = usize> + '_ {
+    offsets.windows(2).map(|w| (w[1] - w[0]) as usize)
```

Comment:
This should be a saturating sub or this will end up crashing the app on malformed inputs.